### ISSUE TITLE
Namespaces and pages show up twice when the namespace is also a page

### ISSUE THREAD

**Issue by cralor** [gturri/nspages#52]
I don't think it should be intended behavior to have the namespace page and namespace in both "Namespaces" and "Pages".

 I cannot currently find the functionality to make this not happen. And the feature to merge Namespaces and Pages continues to have them show up twice.

Thanks for a great extension.

**Comment by gturri** [comment]
I'm not sure I understand what you mean: as far as I can see, when I use just "<nspages>" then I have nothing duplicated: all subnamespaces appear only in the "Namespaces" section, and all the pages in the current one only appear in the "Pages" section.

Could you please give some more details?

**Comment by cralor** [comment]
Thanks for the comment. What do you mean by using just quotations?

**Comment by gturri** [comment]
Sorry, typo, I meant: just "&lt;nspages>"

**Comment by cralor** [comment]
I was looking for a way to have two separate <nspages>. One for namespaces and the other for pages. I was hoping there was a way to take out the "namespace pages" from the pages <nspages>... like a -nonamespaces option.

But anyway, I think I'm going to actually be okay with using the merge option and just display the namespaces and pages together.

Thanks!

**Comment by gturri** [comment]
There is no -nonamespaces option because it's the default behavior.
If you want to have a section for pages and another section for namespaces, you can do:

    Just pages: <nspages>
    Just namespaces: <nspages -subns -nopages>

Does it fit what you want to do?

**Comment by cralor** [comment]
\<nspages\> is not just pages when the namespace itself is a page.

**Comment by gturri** [comment]
Ok, I guess what I don't understand, is the notion of namespace which is itself a page. What do you mean by that. Could you perhaps give an example of page hierarchy that would lead to the case you're describing?

**Comment by cralor** [comment]
Sure.

So if you make a structure like :toys:blocks and do an \<nspages\> from the root, you will have 'toys' listed in both "Namespaces" and "Pages" and 'blocks' listed only in "Pages" like this:

Namespaces
- Toys

Pages
- Toys
- Blocks

**Comment by gturri** [comment]
I tried to reproduce it. I : 
* installed a fresh new wiki and installed the plugin
* went to http://xxx/doku.php?id=toys:blocks and created the page with some dummy content
* went to http://xxx/doku.php?id=start and created the page with "$lt;nspages>"

it only displayed:

Pages in this namespace:
* start

which is, as far as I understand, the expected behavior.

What did I miss in order to reproduce the behavior you're describing?
(Perhaps you could provide a link to a page where I could observe it?)

**Comment by cralor** [comment]
My wiki is behind .htaccess login authentication, so unfortunately it isn't quite easy to show.

What I'm addressing more specifically is if you made the following pages:
http://xxx/doku.php?id=toys:blocks
http://xxx/doku.php?id=toys
http://xxx/doku.php?id=sidebar

Now you're at the homepage, start. On the sidebar you have what I described in the previous comment: Toys in both namespaces and pages.

I think this isn't necessarily a nspages discussion, but rather a dokuwiki design discussion at this point. I realize that "namespaces" and "pages" can become intertwined pretty easily. A namespace can also be a page, and as soon as a page is created "within" a page (by doing xxx:newpagehere) you immediately categorize "xxx" as a new namespace.

I've realized that my end goal is to simply make my first level namespaces not appear under pages, even though they are. This unique ask is something that can be achieved by using nspages' -exclude option. Might be a slight hassle revising for every new top-level namespace, but it's not too bad considering the uniqueness of the situation. And like I mentioned previously, my other option is to simply show them both together.

Thanks for the effort you've put in towards helping me with this issue.
